### ISSUE TITLE
False positive with attrs converter

### ISSUE THREAD

**Issue by tsx** [python/mypy#8625]
Looks like mypy can't infer generic parameter properly when used in attrs converter.

```python
from attr import attrs, attrib
from typing import FrozenSet, Text


@attrs
class Test(object):
    values = attrib(type=FrozenSet[Text], converter=frozenset)


Test(values={'a'})
```

**Actual**: reporting error
```
error: Argument 1 to <set> has incompatible type "str"; expected "_T"
```

**Expected**: no errors.

mypy version 0.770, python 3.5.2, no flags are used.
Latest master has basically the same error message except generic param has a different name:
```
error: Argument 1 to <set> has incompatible type "str"; expected "_T_co"
```

Possibly related: https://github.com/python/mypy/issues/5313

**Comment by msullivan** [comment]
Yeah, that does look wrong. Not sure how easy fixing it would be.

**Comment by markusschmaus** [comment]
This also happens in case of a generic function.
```python
from attr import attrs, attrib
from typing import FrozenSet, Text, Iterable, TypeVar

_T = TypeVar('_T')
def converter(ar: Iterable[_T]) -> FrozenSet[_T]:
    return frozenset(ar)

@attrs
class Test(object):
    values = attrib(type=FrozenSet[Text], converter=converter)


Test(values={'a'})
```

results in an error:

```
error: Type variable "scratch_11._T" is unbound
```


I looked into how this could be fixed and here is what I found out for reference:
The attrs plugin gets called during 

https://github.com/python/mypy/blob/41927cf6dbbc87728eab632486372971c45999cb/mypy/semanal_main.py#L78

At this point the type of the `converter` isn't fully processed so when attrs looks at its type at 

https://github.com/python/mypy/blob/41927cf6dbbc87728eab632486372971c45999cb/mypy/plugins/attrs.py#L98 

`converter.is_generic` returns `False` even though the function is generic. The full type of this function is only established during 

https://github.com/python/mypy/blob/41927cf6dbbc87728eab632486372971c45999cb/mypy/semanal_main.py#L79

Which the attrs plugin has already completed.

**Comment by markusschmaus** [comment]
I opened a PR which fixes this issue for builtin types, including `frozenset`. https://github.com/python/mypy/pull/8876/files

It doesn't fix the issue for generic functions, since their type still contains `UnboundType`s when the attr plugin is run.

**Comment by ryansobol** [comment]
Just ran into this bug. Thank you so much for submitting a PR!
